### ISSUE TITLE
Automatic deployment, and generated app.yaml and Dockerfile should come from gcloud app gen-config

### ISSUE THREAD

**Issue by patflynn** [GoogleCloudPlatform/cloud-code-intellij#451]
We currently package template app.yaml and Dockerfiles ourselves to setup auto deployment and config generation.  This is because gcloud app gen-config creates AE2 style configs which are not available broadly. Once this is fixed we can move to using the gcloud command for generating these files.

**Comment by chanseokoh** [comment]
'gcloud preview app gen-config --custom' currently generates an 'app.yaml' with 'env: 2', which is an expected behavior. Although this 'env: 2' should make 'gcloud' deploy an app to AE MVM, I can see that it currently deploys an app to AE standard. However, I think I can at least start implementing this and merge it later when 'gcloud' is eventually fixed to deploy an app to AE2.

**Comment by patflynn** [comment]
the problem with env:2 is that it was whitelist only last time I checked.

Amir do you know if env:2 is off the whitelist?

**Comment by patflynn** [comment]
@arouzrokh

**Comment by arouzrokh** [comment]
Hey guys, 
we should not, at the moment, use gen-config because it does not support the duality that we are seeing for GAE standard and Flex. The code is also to some extend orphaned as the engineer whom first wrote this code has since departed. 

Please package the correct app.yaml and Dockerfile that is tested to work with the deployment at hand for now until we actually end up having reliable fingerprinting technology, if ever. 

Thanks,
Amir

**Comment by chanseokoh** [comment]
@arouzrokh Can we now use `gcloud gen-config`? When do you think we can use it?

**Comment by etanshaul** [comment]
blocked until gen-config fully supports our requirements

**Comment by arouzrokh** [comment]
@rdayal Rajeev who's the person handling gen-config? it sits with gcloud right?

**Comment by patflynn** [comment]
@meltsufin I believe Mike's last fix should have unblocked this.

**Comment by meltsufin** [comment]
Yes, it has been fixed in gcloud, and should be there in the next release.

**Comment by etanshaul** [comment]
@meltsufin do you know which version of gcloud has this? I'm trying to confirm if its been released or not.

**Comment by meltsufin** [comment]
I'm not sure which version it was added to, but it is there now. Look for `vm: true` in this file `google-cloud-sdk/platform/ext-runtime/java/data/app.yaml.template`.

**Comment by etanshaul** [comment]
closed as "won't fix" due to limited enhancements for ae flex
